Automattic/wp-calypso
# Add sidebar nudge for domain to paid plan

This PR adds a sidebar nudge for a sites on a free plan with a mapped domain.

The a/b test `domainToPaidPlanUpsellNudge` has variants `skip` and `show` to determine which users will see the nudge.

To qualify, a site must have a domain mapping and a free plan.  If the allocated a/b test variant is `show` then the following sidebar nudge will be visible:

![screen shot 2017-05-02 at 2 18 38 pm](https://cloud.githubusercontent.com/assets/1926/25603794/4664bd4c-2f42-11e7-9a1e-c50444246fca.png)

The `calypso_upgrade_nudge_impression` with `{ cta_name: "domain-to-paid-sidebar" }` tracks event will fire when the nudge is seen.  If the user clicks on the nudge then a `calypso_upgrade_nudge_cta_click` with `{ cta_name: "domain-to-paid-sidebar" }` tracks event will fire and the user will be taken to `/plans/<site_slug>`.

# Testing

* Create a site with a mapped domain but the free plan (this can be accomplished by purchasing a plan and registering a domain and then refunding the plan)
* select the 'show' variant for the domainToPaidPlanUpsellNudge test
* on the sidebar for your site, you should see the nudge.

Setting the domainToPaidPlanUpsellNudge a/b test to the skip variant will make the nudge disappear.
